cmd/peer-rest-server.go

```diff
@@ -1137,30 +1137,14 @@ type SpeedtestResult struct {
 	Error     string
 }
 
-// SpeedtestObject implements "random-read" object reader
-type SpeedtestObject struct {
-	buf               []byte
-	remaining         int
-	totalBytesWritten *uint64
+// randomReader returns random data upon every
+// read never returns io.EOF
+type randomReader struct {
+	buf []byte
 }
 
-func (bo *SpeedtestObject) Read(b []byte) (int, error) {
-	if bo.remaining == 0 {
-		return 0, io.EOF
-	}
-	if len(b) == 0 {
-		return 0, nil
-	}
-	if len(b) > len(bo.buf) {
-		b = b[:len(bo.buf)]
-	}
-	if len(b) > bo.remaining {
-		b = b[:bo.remaining]
-	}
-	copy(b, bo.buf)
-	bo.remaining -= len(b)
-
-	atomic.AddUint64(bo.totalBytesWritten, uint64(len(b)))
+func (rr *randomReader) Read(b []byte) (int, error) {
+	copy(b, rr.buf)
```

Comment:
If len(rr.buf) < len(p) the upper part is not written to.

Reply:
Yes but that is allowed @klauspost i.e is it required that we should return exact bytes that was asked ?

I don't that is necessary here.